### PR TITLE
Add changelog lint script

### PR DESCRIPTION
**Pull request by fisker**
So many errors in changelogs

### CODE REVIEW

**Comment by fisker** [comment]
Do you know how to get PR number and commit files on CI, I think it's possible.

**Comment by evilebottnawi** [comment]
Maybe it is in env variables? Need look at documentation, but for `commit files` you need run git commands i think

**Comment by fisker** [comment]
If we can get the PR number, test can be stricter.

**Comment by thorn0** [comment]
> If we can get the PR number, test can be stricter

Let's not overcomplicate things. Sometimes a PR is a fix for another PR in the same release, so they share one changelog entry.

**Comment by fisker** [comment]
Then good to go

File: changelog_unreleased/handlebars/pr-7051.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#### handlebars: Fix superfluous line breaks in ConcatStatement ([7051](https://github.com/prettier/prettier/pull/7051) by [@dcyriller](https://github.com/dcyriller))
+#### handlebars: Fix superfluous line breaks in ConcatStatement ([#7051](https://github.com/prettier/prettier/pull/7051) by [@dcyriller](https://github.com/dcyriller))
```
Comment (thorn0) [inline changelog_unreleased/handlebars/pr-7051.md]: Can you please also validate that headers 1) have four `#`, 2) don't start with redundant prefixes like "handlebars:", "HTML:", etc.?

File: cspell.json [hunk 2]
```diff
@@ -122,6 +122,7 @@
         "gitattributes",
         "githook",
         "gitignore",
+		"gitkeep",
```
Comment (thorn0) [inline cspell.json]: tabs instead of spaces?

File: scripts/lint-changelog.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  }
+}
+
+const authorRegex = /by \[@(.*?)\]\(https:\/\/github\.com\/\1\)/;
+
+const template = fs.readFileSync(
+  path.join(CHANGELOG_ROOT, TEMPLATE_FILE),
+  "utf8"
+);
+const templateComment = template.match(/<!--[\s\S]*?-->/);
```
Comment (thorn0) [inline scripts/lint-changelog.js]: It still works because include converts arrays to string, but you forgot `[0]`.

Comment (fisker) [reply scripts/lint-changelog.js]: fixed https://github.com/prettier/prettier/pull/7320/commits/a6fa99696b76e840e8e8fe40093457b4e40b8e6b

File: cspell.json (hunk 2 above)
Comment (fisker) [reply cspell.json]: fixed https://github.com/prettier/prettier/pull/7320/commits/e8dbd2369d616bdf75a56d47803c7867cd0621a8

File: changelog_unreleased/handlebars/pr-7051.md (hunk 1 above)
Comment (fisker) [reply changelog_unreleased/handlebars/pr-7051.md]: Done.

Comment (thorn0) [reply changelog_unreleased/handlebars/pr-7051.md]: yes

Comment (fisker) [reply changelog_unreleased/handlebars/pr-7051.md]: Done https://github.com/prettier/prettier/pull/7320/commits/baa548761a13638de335d1741a17fe601cf96626

**Pull request merged**